packages/samples/headless-react/src/pages/SamlPage.tsx

```diff
@@ -1,38 +1,41 @@
 import {buildSamlClient, SamlClientOptions} from '@coveo/auth';
 import {buildSearchEngine} from '@coveo/headless';
-import {useEffect, useState, PropsWithChildren} from 'react';
+import {useEffect, useState, PropsWithChildren, useMemo} from 'react';
 import {AppContext} from '../context/engine';
 
-export function SamlPage(props: PropsWithChildren<SamlClientOptions>) {
-  const [accessToken, setAccessToken] = useState('');
-  const saml = buildSamlClient(props);
+export function SamlPage({
+  children,
+  ...samlClientOptions
+}: PropsWithChildren<SamlClientOptions>) {
+  const [initialAccessToken, setInitialAccessToken] = useState('');
+  const samlClient = useMemo(
+    () => buildSamlClient(samlClientOptions),
+    [samlClientOptions]
+  );
 
   useEffect(() => {
-    getToken();
-  }, []);
+    samlClient.authenticate().then(setInitialAccessToken);
+  }, [samlClient]);
```

Comment:
In isolation, we would want to call `authenticate` whenever a new SAML client is built, since we wouldn't know why it was rebuilt. This is the paradigm I tried to apply for every `useMemo`/`useEffect`.